modules/standard/Path.chpl

```diff
@@ -286,4 +286,80 @@ proc file.realPath(): string throws {
       return false;
     }
   }
+
+  /* Determines and returns the longest common path prefix of
+   all the string pathnames provided. If no arguments are specified, returns
+   the empty string.
+
+   :arg paths: Any number of paths as an array
+   :type paths: `array`
+
+   :return: The longest common path prefix
+   :rtype: `string`
+*/
+
+  proc commonPrefix(paths: []): string {
+
+  var result: string = "";    // result string
+  var inputLength = paths.size;   // size of input array
+  var start: int = paths.domain.first;
+  var end: int = paths.domain.last;
+  var firstPath = paths[start];
+  var delimiter: string;
+  var minimum: int;
+  var flag: int = 0;
+
+  // if input is empty, return empty string.
+  // if input is just one string, return that string as longest common prefix path.
+
+  if inputLength == 0 then {
+    return result;
+  } else if inputLength == 1 then{
+    return firstPath;
+  }
+
+  // finding delimiter to split the paths.
+
+  if firstPath.find("\\", 1..firstPath.length) == 0 then {
+    delimiter = "/";
+  } else {
+    delimiter = "\\";
+  }
+
+  var prefixArray = firstPath.split(delimiter, -1, false);    // array of resultant prefix string
+
+  var pos = prefixArray.size;   // rightmost index of common prefix
+  var minPathLength = prefixArray.size;
+
+  for i in (start+1)..end do {
+
+    var tempArray = paths[i].split(delimiter, -1, false);   // temporary array storing the current path under consideration
+
+    minimum = min(prefixArray.size, tempArray.size);
+
+    if minimum < minPathLength then {
+      minPathLength = minimum;
+    }
+
+    for (itr, ind) in zip(1..minimum, 1..minimum) do {
+      if (tempArray[itr]!=prefixArray[ind] && ind<=pos) {
+        pos = ind;
+        flag=1;   // indicating that pos was changed
```

Comment:
If you're not updating the `prefixArray` you could just compare `pos` against its size instead of using the `flag` variable, right?

Reply:
For input like ['/Users/a','/Users/b'], `pos>=prefixArray.size` but removal needs to be done from `pos..prefixArray.size`. Thus, we realise it's a case where `pos` is being updated but due to `pos==prefixArray.size`, it goes to the wrong condition.
So it is now simply checking that if pos was changed, remove starting from `pos` else the strings are subsets so remove starting from `minPathLength`.

Reply:
I think going from pos to prefixArray.size assumes that prefixArray.size == prefixArray.last, but that won't always be the case (you'll need to create an array with a different index set to verify this)

Reply:
You will want to check against prefixArray.domain.last, but perhaps the initial value for pos should be set higher than the highest index to avoid confusion